### PR TITLE
DACCESS-360 - Refactor advanced search builder logic

### PR DESCRIPTION
**Pull request by chrisrlc**
Also addresses several bugs and a feature request related to advanced search (and attempts not to touch basic/simple catalog search at all for now):

- [DACCESS-360 - Use boolean operators instead of prefix operators when building solr queries](https://culibrary.atlassian.net/browse/DACCESS-360)
    - Prefix operators are still being used for some simple catalog searches and bento searches
- [DACCESS-360 - Search terms sent to solr should be wrapped in quotation marks](https://culibrary.atlassian.net/browse/DACCESS-360)
- [DACCESS-333 - Advanced search query for "arctic animals" returns more search results than basic](https://culibrary.atlassian.net/browse/DACCESS-333)
- [DACCESS-283 - Boolean logic for Advanced Search](https://culibrary.atlassian.net/browse/DACCESS-283)
- [DACCESS-205 - Investigate: Why does an all fields search return different # of results in advanced vs catalog?](https://culibrary.atlassian.net/browse/DACCESS-205)
- [DACCESS-354 - 'OR' Search returning 0 results](https://culibrary.atlassian.net/browse/DACCESS-354)
- [DACCESS-288 - Users want to use quotation marks and be able to enter multiple phrases in a single search box.](https://culibrary.atlassian.net/browse/DACCESS-288)

Right now, this just cleans up the code, but it's still not very object-oriented... after we've taken a look at the catalog search builder logic, we should be able to address that.

Also deletes lib/parsing_nesting, since this seems to have been copied directly from the [blacklight_advanced_search](https://github.com/projectblacklight/blacklight_advanced_search/tree/main/lib/parsing_nesting) plugin and is not referenced anywhere in our code.

Now on my dev vm! http://aws-108-106.internal.library.cornell.edu:3000/advanced

TODO

- Will need to sync with Melissa to see if the Advanced Search Tips below the advanced search form needs updating.

### CODE REVIEW

**Review by jgreidy — APPROVED** [review]
So much code! I like the new tests and cleanup.
Were there schema changes in solr?

File: blacklight-cornell/app/models/search_builder.rb [hunk 1]
```diff
@@ -33,62 +33,64 @@ def sortby_callnum user_parameters
...
-                    	end                   
-                    else
-				      fq_string = '{!term f=' + key + '}' + val
-				    end
-				  	user_parameters[:fq] << fq_string
-          	    	blacklight_params[:fq] = user_parameters[:fq]
-          	  	end
-          	 # end
-          	end
+          solr_parameters[:fq] = []
```
Comment (chrisrlc) [inline blacklight-cornell/app/models/search_builder.rb:71]: Note: I did not actually make any changes from here to the end of `#advsearch` except for fixing the padding and renaming `user_parameters` to `solr_parameters` to make it more obvious what this data is. (Just an FYI to make reviewing this PR easier.)

**Comment by chrisrlc** [comment]
> So much code! I like the new tests and cleanup. Were there schema changes in solr?

No schema changes in solr as far as I know! (Though there will be a couple changes needed to support the solr 9 upgrade, but we have separate tickets that will handle that, not included in this branch.)

**Comment by chrisrlc** [comment]
TODO: Wrap `Journal/Periodical` in quotation marks, i.e. `format:"Journal/Periodical"`. Confirmed with Frances that unquoted still works, but if the format ever changed to something with spaces, or another format with spaces was added for a different search field, we'd definitely want format to be quoted. Making format quoted now would future proof this a bit more.

**Pull request merged**